src/tests/ftest/cart/util/cart_logtest.py

```diff
@@ -396,28 +405,23 @@ def _check_pid_from_log_file(self, pid, abort_on_warning, leak_wf, show_memleaks
                             # -DER_NOMEM
                             show = False
                     elif line.rpc:
-                        # Ignore the SWIM RPC opcode, as this often sends RPCs
-                        # that fail during shutdown.
+                        # Ignore the SWIM RPC opcode, as this often sends RPCs that fail during
+                        # shutdown.
                         if line.rpc_opcode == '0xfe000000':
                             show = False
-                    # Disable checking for a number of conditions, either
-                    # because these errors/lines are badly formatted or because
-                    # they're intermittent and we don't want noise in the test
-                    # results.
+                    # Disable checking for a number of conditions, either because these errors/lines
+                    # are badly formatted or because they're intermittent and we don't want noise in
+                    # the test results.
                     if line.fac == 'external':
                         show = False
-                    elif show and server_shutdown and (line.get_msg().endswith(
-                        "DER_SHUTDOWN(-2017): 'Service should shut down'")
-                            or line.get_msg().endswith(
-                                "DER_NOTLEADER(-2008): 'Not service leader'")):
-                        show = False
-                    elif show and line.function == 'rdb_stop':
+                    elif show and server_shutdown and any(map(line.get_msg().endswith,
+                                                              SHUTDOWN_RC)):
```

Comment:
Love using `any(map(...))`

Reply:
I took this from your suggestion on a related but different PR.